apps/api/src/database/repositories/competition-repository.ts

```diff
@@ -24,67 +26,97 @@ import { PartialExcept } from "./types.js";
  * Find all competitions
  */
 export async function findAll() {
-  return await db.query.competitions.findMany();
+  return await db
+    .select({
+      crossChainTradingType: tradingCompetitions.crossChainTradingType,
+      ...getTableColumns(competitions),
+    })
+    .from(tradingCompetitions)
+    .innerJoin(
+      competitions,
+      eq(tradingCompetitions.competitionId, competitions.id),
+    );
```

Comment:
From the perspective of the caller of this function, they don't know there are actually two tables being queried to get the data. But yes, we join the `competitions` and `trading_competitions` tables here to get all the data we want.